provider/azure/internal/azurecli/az.go

```diff
@@ -25,13 +26,39 @@ type AzureCLI struct {
 	Exec func(cmd string, args []string) (stdout []byte, err error)
 }
 
+// Error represents an error returned from the Azure CLI.
+type Error struct {
+	exec.ExitError
+}
+
+// Error implements the error interface.
+func (e *Error) Error() string {
+	if len(e.Stderr) == 0 {
+		return e.ExitError.Error()
+	}
+	n := bytes.Index(e.Stderr, []byte{'\n'})
```

Comment:
IndexByte?

Also, maybe convert to string just once and then use string operations?